sdk/core/Azure.Core/src/Pipeline/HttpClientTransport.cs

```diff
@@ -447,11 +449,40 @@ public PipelineResponse(string requestId, HttpResponseMessage responseMessage, S
                 {
                     // Make sure we don't dispose the content if the stream was replaced
                     _responseMessage.Content = null;
+                    // Make sure we drop our cached value for the content
+                    _content = null;
 
                     _contentStream = value;
                 }
             }
 
+            public override BinaryData Content
+            {
+                get
+                {
+                    if (_content != null)
+                    {
+                        return _content;
+                    }
+
+                    if (ContentStream == null)
+                    {
+                        _content = new BinaryData(Array.Empty<byte>());
```

Comment:
It would be good if we added BinaryData.Empty<T>

Reply:
Will do.